packages/compiler/src/collector/expression.ts

```diff
@@ -1,49 +1,122 @@
-        return bool.getLiteralValue();
+        return types.staticValue(bool.getLiteralValue());
       })
       // horizontal={false}
       // eslint-disable-next-line @typescript-eslint/unbound-method -- FIXME
       .when(Node.isFalseLiteral, (bool) => {
-        return bool.getLiteralValue();
+        return types.staticValue(bool.getLiteralValue());
       })
       // fontSize={['24px', '32px']}
       .when(Node.isArrayLiteralExpression, (array) => {
-        const arrayExpression = array.getElements().map((elm) => {
-          return handleJsxExpression(elm);
-        }) as (string | number | undefined)[];
+        const arrayExpression = array
+          .getElements()
+          .map((elm) => handleJsxExpression(elm))
+          .map((x) =>
+            x?.type === "Static"
+              ? typeof x.value === "string" || typeof x.value === "number"
+                ? x.value
+                : undefined
+              : undefined
+          );
         // If the arrayExpression includes undefined, it means some of its elements
         // could not be statically analyzed, hence handle it at runtime.
-        return arrayExpression.includes(undefined)
+        return arrayExpression.some((x) => x === undefined)
           ? undefined
-          : arrayExpression;
+          : types.staticValue(arrayExpression);
       })
-      // fontSize={{xl: '2rem'}['xl']}
+      // fontSize={someFlag ? "24px" : "32px"}
+      .when(Node.isConditionalExpression, (conditional) => {
+        const condition = conditional.getCondition();
+        const whenTrue = handleJsxExpression(conditional.getWhenTrue());
+        const whenFalse = handleJsxExpression(conditional.getWhenFalse());
+
+        return whenTrue?.type === "Static" && whenFalse?.type === "Static"
+          ? types.conditional({
+            expression: condition.getText(),
+            whenTrue: whenTrue.value,
+            whenFalse: whenFalse.value,
+          })
+          : undefined;
+      })
+      // _hover={{color: 'red'}}
       .when(Node.isObjectLiteralExpression, (obj) => {
-        // TODO
-        return undefined;
+        const entries: [string, types.Value][] = [];
+        for (const prop of obj.getProperties()) {
+          if (Node.isPropertyAssignment(prop)) {
+            const initializer = prop.getInitializer();
+            if (initializer) {
+              const propName = prop.getName();
+              const value = handleJsxExpression(initializer);
+              if (value === undefined) {
+                return undefined;
+              }
+              entries.push([propName, value]);
+            }
+          } else {
+            return undefined;
+          }
+        }
+
+        if (entries.every((e) => e[1].type === "Static")) {
+          return types.staticValue(
+            Object.fromEntries(
+              entries.map((e) =>
+                e[1].type === "Static" ? [e[0], e[1].value] : ({} as never)
```

Comment:
Oh right, actually this is accidental since the argument of staticValue is `StaticValue` but  the type of `Object.fromEntries(entries)` is `{ [k: string]: types.Value; }`. Ideally type check should fail and somehow I thought it fails but it actually passes 🤦 

Anyway the runtime logic is correct so as long as we cover this part in test cases it should be acceptable, thanks for removing redundant checks 👍

Reply:
I also missed this although I wrote it: it's because `e[1].value` is needed to unwrap the value from `Static` variant. `e[1]` is like `{type: "Static", value: /* actual value here */ }`